### ISSUE TITLE
Runtime Error Uncaught (in promise): false on Ionic 2.1.0

### ISSUE THREAD

**Issue by badpenguin** [driftyco/ionic#10588]
**Ionic version:**  (check one with "x")
[ ] **1.x**
[X] **2.x**

**I'm submitting a ...**  (check one with "x")
[X] bug report
[ ] feature request
[ ] support request => Please do not submit support requests here, use one of these channels: https://forum.ionicframework.com/ or http://ionicworldwide.herokuapp.com/

**Current behavior:**

I'm getting this problem too with ionic 2.1.0.
It only happens with the device (so when plugins are working).
It only happens with push or pop of NavCtrl; it will never happen with Loading or Alert because i've already added a .catch() to all of them.
From the original stacktrack it seems to happen inside the transistion code.

Reference:
- Previous bug https://github.com/driftyco/ionic/issues/10046
- https://forum.ionicframework.com/t/is-there-a-way-track-where-the-unhandled-promise-error-has-happened/81200/12

**Expected behavior:**

At least a way to track wich is the code causing the error without referencing polyfill.js

**Steps to reproduce:**
Sorry, its very hard to reproduce. It only happens when doing testing on real devices.
Using livereload increase the chance of getting it.

**Other information:**

stack trace:
```
"Error: Uncaught (in promise): false
02-27 22:28:06.047 32263 32263 I chromium:     at s (/build/polyfills.js:3:4211)
02-27 22:28:06.047 32263 32263 I chromium:     at s /build/polyfills.js:3:4034)
02-27 22:28:06.047 32263 32263 I chromium:     at /build/polyfills.js:3:4574
02-27 22:28:06.047 32263 32263 I chromium:     at t.invokeTask /build/polyfills.js:3:9723)
02-27 22:28:06.047 32263 32263 I chromium:     at Object.onInvokeTask (/build/main.js:42519:37)
02-27 22:28:06.047 32263 32263 I chromium:     at t.invokeTask (/build/polyfills.js:3:9659)
02-27 22:28:06.047 32263 32263 I chromium:     at e.runTask (/build/polyfills.js:3:7083)
02-27 22:28:06.047 32263 32263 I chromium:     at i (/build/polyfills.js:3:3671)", source: /__ion-dev-server/ion-dev.js?v=1.1.4 (156)
```

ionic info:
```
Cordova CLI: 6.5.0 
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
ios-deploy version: Not installed
ios-sim version: Not installed
OS: Linux 4.8
Node Version: v6.10.0
Xcode version: Not installed
```

package.json:
```
"dependencies": {
    "@angular/common": "2.2.1",
    "@angular/compiler": "2.2.1",
    "@angular/compiler-cli": "2.2.1",
    "@angular/core": "2.2.1",
    "@angular/forms": "2.2.1",
    "@angular/http": "2.2.1",
    "@angular/platform-browser": "2.2.1",
    "@angular/platform-browser-dynamic": "2.2.1",
    "@angular/platform-server": "2.2.1",
    "@ionic/storage": "1.1.7",
    "ionic-angular": "2.1.0",
    "ionic-native": "2.5.1",
    "ionicons": "3.0.0",
    "rxjs": "5.0.0-beta.12",
    "sw-toolbox": "3.4.0",
    "zone.js": "0.6.26"
  },
  "devDependencies": {
    "@ionic/app-scripts": "1.1.4",
    "typescript": "2.0.9"
  }
```

**Comment by jgw96** [comment]
Thanks, we will look into this.

**Comment by theHellyar** [comment]
Please do look into this....getting same error....
Cordova CLI: 6.4.0 
Ionic Framework Version: 2.1.0
Ionic CLI Version: 2.2.1
Ionic App Lib Version: 2.2.0
Ionic App Scripts Version: 1.0.0
ios-deploy version: 1.9.0 
ios-sim version: 3.1.1 
OS: macOS Sierra
Node Version: v7.2.1
Xcode version: Xcode 8.2 Build version 8C38

**Comment by Hodes** [comment]
I have posted a comment on another issue that refers to the same problem I think, please check the comment: 

[https://github.com/driftyco/ionic/issues/10046#issuecomment-285122667](https://github.com/driftyco/ionic/issues/10046#issuecomment-285122667)

Thanks !

**Comment by kleeb** [comment]
was working on Ionic 2.3.0.
Appeared again on 3.0.0-beta.3 doing dismiss() on Loading object

**Comment by AleFons** [comment]
Can confirm; it's happening on 3.0.0-beta.3 and 3.0.0 final.

**Comment by kleeb** [comment]
The case is that the 3.0.0 bundle is way faster than 2.x, and some components are being dismissed faster than they actually arrive in the controllers. 
We slightly refactored our LoadingControllers, that we started always dismissing them inside promises of present. Before dismiss was called before present finished. I guess its similar situation for other cases in this issue.

**Comment by chandanch** [comment]
I facing the same issue too: 

```typescript
let loading = loadingController.create({
  spinner : 'hide',
  content : `Loading...`
});
loading.present().then(() => {
  //do some work
});
//Do some more work
loading.dismiss();
```

**Ionic info:**
`
Cordova CLI: 6.5.0 
Ionic Framework Version: 3.0.1
Ionic CLI Version: 2.2.2
Ionic App Lib Version: 2.2.1
Ionic App Scripts Version: 1.3.0
ios-deploy version: Not installed
ios-sim version: 5.0.8 
OS: macOS Sierra
Node Version: v6.10.1
Xcode version: Xcode 8.2.1 Build version 8C1002
`

**Comment by kleeb** [comment]
@chandanch 

dismiss should be inside the Promise like that:

```
loading.present().then(() => {
  //do some work
  loading.dismiss();
});

```

**Comment by chandanch** [comment]
@kleeb I added the dismiss within the promise but still same error:

```typescript
loading.present().then(() => {
    //do some work
    loading.dismiss();
})
```

**Comment by Ross-Rawlins** [comment]
also getting this error

**Comment by Fearful** [comment]
also getting this error
ionic v3

**Comment by badpenguin** [comment]
The only solution i found its to add a ".catch(logErrors)" everywhere, at every function. Where logErrors its just a shortcut for console.warning() or other things.

**Comment by chandanch** [comment]
This also happens if you are trying to dismiss the already dismissed overlay component

**Comment by PurpleEdge2214** [comment]
Wrapping the showLoading function in a wrapper seems to help...

```
  showLoading(message: string) {
    return new Promise(resolve => {
      this.loading = this.loadingCtrl.create({
        content: message,
        //dismissOnPageChange: true,  // bug, causes exception
        spinner: 'crescent'
      })
      this.loading.present()
        .then(() => {

          setTimeout(() => {
            this.loading.dismiss();
          }, 5000);

          resolve('loaded');
        })
    })
  }

```

**Comment by neo-split** [comment]
Im also facing same issue here. Looking forward to a fix / hack!

**Comment by PurpleEdge2214** [comment]
Read the comments above for an explanation. You are getting the error because the loading is being dismissed before it has finished loading. If you place the call to show the loading within a promise, and wait for the loading to complete before you continue your code you will most likely avoid the error.

**Comment by manucorporat** [comment]
This should be fixed in 3.0.1

**Comment by ifelere** [comment]
I saw the same error one of  my projects a couple of hours back. It was caused by me calling actionSheet.dismiss() inside button handlers.
